src/routes/_private/flights/-components/upsert-flights-dialog.tsx

```diff
@@ -0,0 +1,260 @@
+    { airlineId: selectedAirlineId?.toString() || "", page: 1 },
+    { enabled: !!selectedAirlineId },
+  );
+
+  const airlineCities = useMemo(() => {
+    return Array.isArray(airlineCitiesData?.data) ? airlineCitiesData.data : [];
+  }, [airlineCitiesData]);
+
+  const {
+    clearErrors,
+    formState: { errors },
+    handleSubmit,
+    register,
+    reset,
+    setError,
+    setValue,
+    watch,
+  } = useForm<CreateFlightPayload>({
+    mode: "onTouched",
+    resolver: zodResolver(getFlightSchema()),
+    defaultValues: {
+      airline_id: initialAirlineId,
+      departure_city_id: initialDepartureId,
+      arrival_city_id: initialArrivalId,
+      departure_time: toDateInput(currentFlight?.departureTime),
+      arrival_time: toDateInput(currentFlight?.arrivalTime),
+    },
+  });
+
+  useEffect(() => {
+    if (isOpen && currentFlight && !hasResetRef.current) {
+      reset({
+        airline_id: initialAirlineId,
+        departure_city_id: initialDepartureId,
+        arrival_city_id: initialArrivalId,
+        departure_time: toDateInput(currentFlight?.departureTime),
+        arrival_time: toDateInput(currentFlight?.arrivalTime),
+      });
+      setSelectedAirlineId(initialAirlineId);
+      hasResetRef.current = true;
+    }
+    if (!isOpen) {
+      hasResetRef.current = false;
+    }
+  }, [isOpen, currentFlight, reset, initialAirlineId, initialArrivalId, initialDepartureId]);
+
+  const handleCreate = (payload: CreateFlightPayload) => {
+    createFlight(payload, {
+      onSuccess: () => {
+        toast.success(t("flights.create.success"));
+        onOpenChange(false);
+        reset();
+      },
+      onError: (error) => {
+        handleAxiosFieldErrors<CreateFlightPayload>(error, setError, t("flights.create.error"));
+      },
+    });
+  };
+
+  const handleUpdate = (payload: UpdateFlightPayload) => {
+    updateFlight(payload, {
+      onSuccess: () => {
+        toast.success(t("flights.update.success"));
+        onOpenChange(false);
+        reset();
+      },
+      onError: (error) => {
+        handleAxiosFieldErrors<UpdateFlightPayload>(error, setError, t("flights.update.error"));
+      },
+    });
+  };
+
+  const onSubmit: SubmitHandler<CreateFlightPayload> = (data) => {
+    const payload = {
+      airline_id: data.airline_id,
+      departure_city_id: data.departure_city_id,
+      arrival_city_id: data.arrival_city_id,
+      departure_time: data.departure_time,
+      arrival_time: data.arrival_time,
+    };
+
+    if (isNewFlight) {
+      return handleCreate(payload);
+    }
+
+    return handleUpdate({ id: currentFlight!.id, ...payload });
+  };
+
+  const handleAirlineChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const id = Number(e.target.value);
+    setSelectedAirlineId(id);
+    setValue("airline_id", id);
+    setValue("departure_city_id", 0);
+    setValue("arrival_city_id", 0);
```

Comment:
isn't there a `clear()` or `reset()` or something like that? that way you can clear all the fields, and just set the value of the airline_id